src/framework/COScript.m

```diff
@@ -405,65 +405,111 @@ - (void)deleteObjectWithName:(NSString*)name {
     [_mochaRuntime removeObjectWithName:name];
 }
 
+# pragma mark - require
+
+// This aims to implement the require resolution algorithm from NodeJS.
+// Given a `module` string required by a given script at the `currentURL`
+// using `require('./path/to/module')`, this method returns a URL
+// corresponding to the `module`.
+// `module` could also be the name of a core module that we shipped with the app
+// (for example `util`), in which case, it will return the URL to that one
+// and set `isRequiringCore` to YES.
+- (NSURL*)resolveModule:(NSString*)module currentURL:(NSURL*)currentURL isRequiringCoreModule: (BOOL*)isRequiringCore {
+    if (![module hasPrefix: @"."] && ![module hasPrefix: @"/"] && ![module hasPrefix: @"~"]) {
+        *isRequiringCore = YES;
+        return self.coreModuleMap[module];
+    }
+    
+    *isRequiringCore = NO;
+    
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    BOOL isRelative = [module hasPrefix: @"."];
+    NSString *modulePath = [module stringByStandardizingPath];
+    NSURL *moduleURL = isRelative ? [NSURL URLWithString:modulePath relativeToURL:currentURL] : [NSURL fileURLWithPath:modulePath];
+    BOOL isDir;
+    
+    if ([fileManager fileExistsAtPath:moduleURL.path isDirectory:&isDir]) {
+        if (!isDir) {
+            // if the module is a proper path to a file, just use it
+            return moduleURL;
+        }
+        // if it's a path to a directory, let's try to find a package.json
+        NSURL *packageJSONURL = [moduleURL URLByAppendingPathComponent:@"package.json"];
+        NSString *jsonString = [[NSString alloc] initWithContentsOfFile:packageJSONURL.path encoding:NSUTF8StringEncoding error:nil];
+        if (jsonString != nil) {
+            NSData *jsonData = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
+            id packageJSON = [NSJSONSerialization JSONObjectWithData:jsonData options:NSJSONReadingMutableLeaves error:nil];
```

Comment:
Sorry I'm being such a pain, but furthermore when I read this through I realise it seems quite odd to go:

1. Read in a string
2. Convert that string to data
3. Parse that data as JSON

It would be a lot more straightforward and more efficient to go:

1. Read in data
2. Parse as JSON